enricher/api/src/main/java/io/fabric8/maven/enricher/api/BaseEnricher.java

```diff
@@ -133,8 +134,30 @@ protected void addInitContainer(PodTemplateSpecBuilder builder, JSONObject initC
         String initContainerAnnotation = builder.buildMetadata().getAnnotations().get(INIT_CONTAINER_ANNOTATION);
         JSONArray initContainers = Strings.isNullOrBlank(initContainerAnnotation) ? new JSONArray()
                 : new JSONArray(initContainerAnnotation);
-        initContainers.put(initContainer);
 
+        // lets avoid duplicates being added
+        boolean duplicate = false;
+        int length = initContainers.length();
+        for (int i = 0; i < length; i++) {
+            JSONObject obj = initContainers.getJSONObject(i);
+            if (duplicateInitcontainer(obj, initContainer)) {
+                duplicate = true;
+            }
+
+        }
+        if (!duplicate) {
+            initContainers.put(initContainer);
+        }
         builder.editMetadata().addToAnnotations(INIT_CONTAINER_ANNOTATION, initContainers.toString()).endMetadata();
     }
+
+    private boolean duplicateInitcontainer(JSONObject a, JSONObject b) {
+        Object name1 = a.get("name");
+        Object name2 = b.get("name");
+        if (Objects.equal(name1, name2)) {
+            log.warn("Found duplicate init containers with names " + name1 + " so ignoring " + b);
```

Comment:
@jstrachan would be really happy to help in avoiding such situations. Is there anything we should test regularly before doing a f-m-p release w.r.t to fabric8-platform ?

There was an issue in the fabric8-platform lately which lead to method mismatch errors because a dependency added to the plugin used a different kubernetes-model than the plugin itself (I documented that in the corresponding PR in fabric8-platform). This what *not* something f-m-p broke, but the dependency added to the plugin deps. I introduced a fix (moving the deps to regular project deps which also works for the dependency enricher) but could test this only superficially (whether the yaml files were created). However it seems, that this was not good enough. Is there any other rigorous test I could perform ? If so, I would run it (even manually) before each f-m-p release.